src/crimson/os/seastore/seastore_types.h

```diff
@@ -468,6 +584,48 @@ struct paddr_t {
   bool is_zero() const {
     return get_device_id() == DEVICE_ID_ZERO;
   }
+  paddr_t add_relative(paddr_t o) const {
+    assert(o.is_relative());
+    common_addr_t ret =
+      dev_addr &
+      ~static_cast<common_addr_t>(
+	std::numeric_limits<device_seastore_off_t>::max());
+    seastore_off_t l = static_cast<seastore_off_t>(dev_addr &
+      static_cast<common_addr_t>(
+	std::numeric_limits<device_seastore_off_t>::max()));
+    seastore_off_t r = static_cast<seastore_off_t>(o.dev_addr &
+      static_cast<common_addr_t>(
+	std::numeric_limits<device_seastore_off_t>::max()));
```

Comment:
This implementation is confusing.  o is being treated as though it is a blk_paddr_t, but relative addrs are (for the moment, at least) always seg_paddr_t.